examples/chip-tool/commands/clusters/ModelCommand.cpp

```diff
@@ -73,3 +74,31 @@ void ModelCommand::Shutdown()
 
     CHIPCommand::Shutdown();
 }
+
+void ModelCommand::CheckPeerICDType()
+{
+    if (mIsPeerLIT.HasValue())
+    {
+        ChipLogProgress(chipTool, "Peer ICD type is set to %s", mIsPeerLIT.Value() == 1 ? "LIT-ICD" : "non LIT-ICD");
+        return;
+    }
+
+    app::ICDClientInfo info;
+    auto destinationPeerId = chip::ScopedNodeId(mDestinationId, CurrentCommissioner().GetFabricIndex());
+    auto iter              = CHIPCommand::sICDClientStorage.IterateICDClientInfo();
+
+    if (iter == nullptr)
+    {
+        return;
+    }
+
+    while (iter->Next(info))
+    {
+        if (ScopedNodeId(info.peer_node.GetNodeId(), info.peer_node.GetFabricIndex()) == destinationPeerId)
+        {
+            ChipLogProgress(chipTool, "Peer is a registered LIT ICD.");
+            mIsPeerLIT.SetValue(true);
+            return;
```

Comment:
The `iter` is leaked in that function. That's the reason why Tizen QEMU tests fails. After fixing the leak CI will pass.

How, about adding a wrapper for `iter`, so it can be managed by RAII? The leak is present not only here... The `src/controller/java/CHIPDeviceController-JNI.cpp` also leaks iterator in case of error path.

Reply:
Thanks, would update it.